enhancements/two-node-fencing/tnf.md

```diff
@@ -0,0 +1,767 @@
+
+#### Infrastructure API Changes
+
+A mechanism is needed for components of the cluster to understand that this is a two-node control-plane topology that may require different handling.
+We will define a new value for the `infrastructureTopology` field of the Infrastructure config's `TopologyMode` enum.
+Specifically, the value of `DualReplica` will be added to the currently supported list, which includes `HighlyAvailable`, `SingleReplica`, and `External`.
+
+InfrastructureTopology is assumed to be immutable by components of the cluster. While there is strong interest in changing this to allow for topology
+transitions to and from TNF, this is beyond the scope of this enhancement proposal and should be detailed in its own enhancement.
+
+#### etcd Operator Changes
+
+Initially, the creation of an etcd cluster will be driven in the same way as other platforms.
+Once the cluster has two members, the etcd daemon will be removed from the static pod definition and recreated as a resource controlled by RHEL-HA.
+At this point, the cluster-etcd-operator (CEO) will be made aware of this change so that some membership management functionality that is now handled by RHEL-HA can be disabled.
+This will be achieved by having the same entity that drives the configuration of RHEL-HA use the OpenShift API to update a field in the CEO's `ConfigMap` - which can only succeed if the control-plane is healthy.
+
+To enable this flow, we propose the addition of a `managedEtcdKind` field which defaults to `Cluster` but will be set to `External` during installation, and will only be respected if the `Infrastructure` CR's `TopologyMode` is `DualReplicaTopologyMode`.
+This will allow the use of a credential scoped to `ConfigMap`s in the `openshift-etcd-operator` namespace, to make the change.
+
+The plan is for this to be changed by one of the nodes during pacemaker initialization. Pacemaker initialization should be initiated by CEO when it detects that the cluster controlPlane topology is set to `DualReplica`.
+
+While set to `External`, CEO will still need to render the configuration for the etcd container in a place where it can be consumed by pacemaker. This ensures that the etcd instance managed by pacemaker can be updated
+accordingly in the case of a upgrade event or whenever certificates are rotated.
+
+#### Install Config Changes
+
+In order to initialize pacemaker with valid fencing credentials, they will be consumed by the installer via the installation config and created on the cluster as a cluster secret.
+
+A sample install-config.yaml for `platform: none` type clusters would look like this:
+```
+apiVersion: v1
+baseDomain: example.com
+compute:
+- name: worker
+  replicas: 0
+controlPlane:
+  name: master
+  replicas: 2
+metadata:
+  name: <cluster-name>
+platform:
+  none:
+    fencingCredentials:
+      bmc:
+          address: ipmi://<out_of_band_ip>
```

Comment:
We are only going to support redfish here I thought?

Reply:
For TP, redfish only. For future releases, we plan on adding IPMI.
I'll update the example.